packages/design-system/src/components/web-components/ds-date-field/ds-date-field.tsx

```diff
@@ -0,0 +1,95 @@
+import { define } from '../preactement/define';
+import { SingleInputDateFieldProps } from '../../DateField/SingleInputDateField';
+import { SingleInputDateField } from '../../DateField';
+import { parseBooleanAttr, parseDateAttr, parseIntegerAttr } from '../wrapperUtils';
+
+const attributes = [
+  'name',
+  'root-id',
+  'inversed',
+  'value',
+  'default-month',
+  'from-date',
+  'from-month',
+  'from-year',
+  'to-date',
+  'to-month',
+  'to-year',
+  'hint',
+  'label',
+  'error-message',
+] as const;
+
+type IncompatibleProps =
+  | 'inversed'
+  | 'defaultMonth'
+  | 'fromDate'
+  | 'fromMonth'
+  | 'fromYear'
+  | 'toDate'
+  | 'toDate'
+  | 'toMonth'
+  | 'toYear';
+
+interface WrapperProps extends Omit<SingleInputDateFieldProps, IncompatibleProps> {
+  name: string;
+  rootId: string;
+  inversed?: string;
+  value?: string;
+  defaultMonth?: string;
+  fromDate?: string;
+  fromMonth?: string;
+  fromYear?: string;
+  toDate?: string;
+  toMonth?: string;
+  toYear?: string;
+}
+
+const Wrapper = ({
+  name,
+  rootId,
+  inversed,
+  value,
+  defaultMonth,
+  fromDate,
+  fromMonth,
+  fromYear,
+  toDate,
+  toMonth,
+  toYear,
+  ...otherProps
+}: WrapperProps) => {
+  return (
+    <SingleInputDateField
+      name={name}
+      id={rootId}
+      inversed={parseBooleanAttr(inversed)}
+      value={value}
+      defaultMonth={parseDateAttr(defaultMonth)}
+      fromDate={parseDateAttr(fromDate)}
+      fromMonth={parseDateAttr(fromMonth)}
+      fromYear={parseIntegerAttr(fromYear)}
+      toDate={parseDateAttr(toDate)}
+      toMonth={parseDateAttr(toMonth)}
+      toYear={parseIntegerAttr(fromYear)}
+      {...otherProps}
+    />
+  );
+};
+
+/* eslint-disable @typescript-eslint/no-namespace */
+declare global {
+  namespace JSX {
+    interface IntrinsicElements {
+      'ds-date-field': JSX.IntrinsicElements['div'] & {
+        [K in (typeof attributes)[number]]?: string;
+      };
+    }
+  }
+}
+/* eslint-enable */
+
+define('ds-date-field', () => Wrapper, {
+  attributes,
+  events: ['onChange', 'onBlur'],
```

Comment:
Unfortunately there's no real information being passed to the `ds-change` event here because the `onChange` handler for `SingleInputDateField` doesn't actually get an event like many other `onChange` handlers do in the design system. [What actually gets passed](https://github.com/CMSgov/design-system/blob/tamara/add-ds-date-field-web-component/packages/design-system/src/components/DateField/SingleInputDateField.tsx#L42) are two strings. The default event translator in our `define` function expects an event. That means you'll either need to create a custom handler here [like we do for ds-pagination](https://github.com/CMSgov/design-system/blob/tamara/add-ds-date-field-web-component/packages/design-system/src/components/web-components/ds-pagination/ds-pagination.tsx#L75), or the `SingleInputDateField` API needs to change.

This lack of information is what's causing the following buggy behavior in the Storybook story:

![Choosing a date from the calendar fails to update the input](https://github.com/user-attachments/assets/b3659cf2-6f53-4c20-b77b-bd22c5f885bf)

Reply:
Even though most of the other web component event-handler unit tests aren't this thorough, we may want to consider checking what's passed to the event handler in the unit tests because it's so different from the others. It looks like we do check for the actual information on `detail` in the `ds-pagination` unit tests.